keps/sig-node/127-user-namespaces/README.md

```diff
@@ -0,0 +1,437 @@
+it'll have limited impact as it'll be running as an unprivileged user there.
+
+The following security vulnerabilities are (completely or partially) mitigated
+with user namespaces as we propose here and it is expected that it will mitigate
+similar future vulnerabilities too.
+- [CVE-2019-5736](https://nvd.nist.gov/vuln/detail/CVE-2019-5736): Host runc binary can be overwritten from container. Completely mitigated with userns.
+  - Score: [8.6 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2019-5736)
+  - https://github.com/opencontainers/runc/commit/0a8e4117e7f715d5fbeef398405813ce8e88558b
+- [Azurescape](https://unit42.paloaltonetworks.com/azure-container-instances/):
+  Completely mitigated with userns, at least as it was found (needs CVE-2019-5736). This is the **first cross-account container takeover in the public cloud**.
+- [CVE-2021-25741](https://github.com/kubernetes/kubernetes/issues/104980): Mitigated as root in the container is not root in the host
+  - Score: [8.1 (HIGH) / 8.8 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2021-25741)
+- [CVE-2017-1002101](https://github.com/kubernetes/kubernetes/issues/60813): mitigated, idem
+  - Score: [9.6 (CRITICAL) / 8.8 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2017-1002101)
+- [CVE-2021-30465](https://github.com/opencontainers/runc/security/advisories/GHSA-c3xm-pvg7-gh7r): mitigated, idem
+  - Score: [8.5 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2021-30465)
+- [CVE-2016-8867](https://nvd.nist.gov/vuln/detail/CVE-2016-8867): Privilege escalation inside containers
+  - Score: [7.5 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2016-8867)
+  - https://github.com/moby/moby/issues/27590
+- [CVE-2018-15664](https://nvd.nist.gov/vuln/detail/CVE-2018-15664): TOCTOU race attack that allows to read/write files in the host
+  - Score: [7.5 (HIGH)](https://nvd.nist.gov/vuln/detail/CVE-2018-15664)
+  - https://github.com/moby/moby/pull/39252
+
+### Goals
+
+Here we use UIDs, but the same applies for GIDs.
+
+- Increase node to pod isolation by mapping user and group IDs
+  inside the container to different IDs in the host. In particular, mapping root
+  inside the container to unprivileged user and group IDs in the node.
+- Increase pod to pod isolation by allowing to use non-overlapping mappings
+  (UIDs/GIDs) whenever possible. IOW, if two containers runs as user X, they run
+  as different UIDs in the node and therefore are more isolated than today.
+  See phase 3 for limitations.
+- Allow pods to have capabilities (e.g. `CAP_SYS_ADMIN`) that are only valid in
+  the pod (not valid in the host).
+- Benefit from the security hardening that user namespaces provide against some
+  of the future unknown runtime and kernel vulnerabilities.
+
+### Non-Goals
+
+- Provide a way to run the kubelet process or container runtimes as an
+  unprivileged process. Although initiatives like
+  [usernetes](https://github.com/rootless-containers/usernetes) and this KEP
```

Comment:
nit: linking this KEP might be better than linking Usernetes
https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2033-kubelet-in-userns-aka-rootless

Reply:
Thanks!